altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -1,65 +1,348 @@
+     */
+    public void spawnInArea(Vector3 leftCorner, List<MapData> maps, float horLength, float verLength)
     {
-        var MapToSpawn = Instantiate(map, position, Quaternion.identity).GetComponent<MapClass>();
-        MapToSpawn.copyFrom(mapData);
-        MapToSpawn.GetComponent<Renderer>().material.mainTexture = mapData.texture;
+        float x = leftCorner.x;
+        float z = leftCorner.z;
+        float y = leftCorner.y;
+
+        foreach (MapData map in maps)
+        {
+            float newX = x - Random.Range(0, horLength);
+            float newZ = z + Random.Range(0, verLength);
+            spawnMap(new Vector3(newX, y, newZ), map);
+        }
     }
 
-    public void spawnAsStack(Vector3 position, MapClass[] mapData) {
+    /**
+     * Will spawn given maps as a stack at a given position.
+     * position: the position the map object should be spawned in the scene
+     * mapData: the data the map object will have
+     * category: the drawer which contains all maps of this category.
+     * */
+    public void spawnAsStack(Vector3 position, MapData[] mapData, string category) {
         for (int i = 0; i < mapData.Length; i++)
         {
-            spawnMap(position, mapData[i]);
+            position.y = position.y + MAPS_Y_OFFSET;
+            spawnMap(position, mapData[i], category);
         }
     }
 
-    public void spawnStacksInCorner(Vector3 leftCorner, float horLength, float verLength, MapClass[][] stacks) {
-        float x = leftCorner.x;
-        float z = leftCorner.z;
-        float y = leftCorner.y;
-        for (int i = 0; i < 2; i++) {
-            for (int k = 0; k < 2; k++) {
-                if (2 * i + k < stacks.Length) {
-                    float newX = x - i * horLength;
-                    float newZ = z + k * verLength;
-                    spawnAsStack(new Vector3(newX, y, newZ), stacks[2 * i + k]);
-                }
-            }
+    /**
+     * Will spawn given maps as a stack in the drawer
+     * position: the position the map object should be spawned in the scene
+     * mapData: the data the map object will have
+     * category: the drawer which contains all maps of this category
+     * */
+    public void spawnAsStack(Vector3 position, List<MapData> mapData, string category)
+    {
+        foreach (MapData map in mapData)
+        {
+            position.y = position.y + MAPS_Y_OFFSET;
+            spawnMap(position, map, category);
+        }
+    }
+
+    /**
+     * Will spawn multiple map stacks aligned in a row 
+     * horLength: width of area
+     * verLength: height of area
+     * stacks: stacks of maps to spawn, first dimension: category, second: subcategories belonging to category
+     * category: drawer which its category is assigned to where row should spawn 
+     * */
+    public void spawnStacksInRow(float horLength, float verLength, MapData[][] stacks, string category) {
+        float x = getDrawerVectorByCategory(category).x;
```

Comment:
Why do we need to define the floats? You're only using them once each so we can spare 3 lines

Reply:
could you further elaborate what you mean?

Reply:
i find it better to read but i can change that